Source/ACE/Entity/Player.cs

```diff
@@ -1633,20 +1692,157 @@ public void HandleActionPutItemInContainer(ObjectGuid itemGuid, ObjectGuid conta
                     var motion = new UniversalMotion(MotionStance.Standing);
 
                     CurrentLandblock.EnqueueBroadcast(Location, Landblock.MaxObjectRange,
-                            new GameMessageUpdateMotion(Guid,
-                                                        Sequences.GetCurrentSequence(SequenceType.ObjectInstance),
-                                                        Sequences, motion));
+                        new GameMessageUpdateMotion(Guid,
+                            Sequences.GetCurrentSequence(SequenceType.ObjectInstance),
+                            Sequences, motion));
                     // CurrentLandblock.EnqueueBroadcast(self shame);
                 }
             });
 
             // Set chain to run
-            putItemInContainerChain.EnqueueChain();
+            pickUpItemChain.EnqueueChain();
+        }
+
+        private void SetInventoryForWorld(ObjectGuid containerGuid, WorldObject inventoryItem)
+        {
```

Comment:
This code should definitely live in WorldObject.  Also note, containerGuid isn't used at all - remove it or use it.